includes/payment-tokens/class-wc-payment-token-cc.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+/**
+ * WooCommerce Credit Card Payment Token
+ *
+ * Representation of a payment token for credit cards
+ * Extends from WC_Payment_Token_eCheck since both types contain last4 digits.
+ *
+ * @class 		WC_Payment_Token_CC
+ * @since		2.6.0
+ * @category 	PaymentTokens
+ * @package 	WooCommerce/PaymentTokens
+ * @author		WooThemes
+ */
+class WC_Payment_Token_CC extends WC_Payment_Token_eCheck {
```

Comment:
The validation could be moved up into the abstract class, and then the CC/eCheck classes can call that and do their own method-specific validations (e.g. an eCheck class might validate routing/account numbers)